### ISSUE TITLE
[FEATURE REQUEST] Pretty string representations

### ISSUE THREAD

**Issue by btjanaka** [icaros-usc/pyribs#125]
### Description

Just like PyTorch's nn.Module, it would be nice if asking for str(archive), str(emitter), str(optimizer) returns a nice string representation of the object rather than Python's default one.

**Comment by btjanaka** [comment]
This would still be nice to have!

**Comment by btjanaka** [comment]
Some references:
- [Python docs for `__repr__`](https://docs.python.org/3/reference/datamodel.html#object.__repr__)
- [UCI ICS 33 lecture notes on `__repr__` and `__str__`](https://ics.uci.edu/~pattis/ICS-33/lectures/operatoroverloading1.txt)
- [PyTorch's implementation of `__repr__`](https://github.com/pytorch/pytorch/blob/v2.12.0/torch/nn/modules/module.py#L2993) -- it seems PyTorch just implements `__repr__` and lets that be called by `__str__`.

In general, `__repr__` is supposed to be a valid Python expression that can be passed to `eval()` to reconstruct an object. However, this can be quite difficult to achieve when you have a ton of internal info to consider (like neural network weights or archive solutions), so PyTorch just gives a representation that would (roughly) allow you to construct a newly initialized version of the object.

**Comment by btjanaka** [comment]
My conclusion is that, at most, we can implement `__str__` on the `Scheduler` (and maybe `BanditScheduler`), since the scheduler is a component that users commonly interact with. Adding `__str__` or `__repr__` to all the archives and emitters is a lot of effort, given the large number of arguments in each class. Furthermore, it would be an extra burden whenever new classes are added.
